test/unittests/execute_test.cpp

```diff
@@ -1135,6 +1135,50 @@ TEST(execute, imported_functions_call_indirect)
     EXPECT_RESULT(execute(instance, 3, {2, 50}), 7);   // isqrt(50)
 }
 
+TEST(execute, imported_function_from_another_module)
+{
+    /* wat2wasm
+    (module
+      (func $sub (param $lhs i32) (param $rhs i32) (result i32)
+        get_local $lhs
+        get_local $rhs
+        i32.sub)
+      (export "sub" (func $sub))
+    )
+    */
+    const auto bin1 = from_hex(
+        "0061736d0100000001070160027f7f017f030201000707010373756200000a09010700200020016b0b");
+    const auto module1 = parse(bin1);
+    auto instance1 = instantiate(module1);
+
+    /* wat2wasm
+    (module
+      (func $add (import "m1" "add") (param $lhs i32) (param $rhs i32) (result i32))
+
+      (func $main (param i32) (param i32) (result i32)
+        get_local 1
+        get_local 0
+        call $add
+      )
+    )
+    */
+    const auto bin2 = from_hex(
+        "0061736d0100000001070160027f7f017f020a01026d31036164640000030201000a0a0108002001200010000"
+        "b");
+    const auto module2 = parse(bin2);
+
+    const auto func_idx = fizzy::find_exported_function(module1, "sub");
+    ASSERT_TRUE(func_idx.has_value());
+
+    auto add = [&instance1, func_idx](Instance&, std::vector<uint64_t> args) -> execution_result {
+        return fizzy::execute(instance1, *func_idx, std::move(args));
```

Comment:
Eventually what would be nice is having a generic function of an export which can be passed here (and the interpreter could optimize it). But that is part of the external API work later.